inject-generator/src/main/java/io/avaje/inject/generator/ProcessingContext.java

```diff
@@ -34,19 +34,20 @@ private ProcessingContext() {}
   private static Filer filer;
   private static Elements elementUtils;
   private static Types typeUtils;
-  private static Set<String> uniqueModuleNames = new HashSet<>();
-  private static Set<String> providedTypes = new HashSet<>();
+  private static final Set<String> uniqueModuleNames = new HashSet<>();
+  private static final Set<String> providedTypes = new HashSet<>();
   private static final Set<String> optionalTypes = new LinkedHashSet<>();
+  private static boolean useAvajeConfig;
 
  static void init(ProcessingEnvironment env, Set<String> moduleFileProvided) {
     processingEnv = env;
     messager = processingEnv.getMessager();
     filer = processingEnv.getFiler();
     elementUtils = processingEnv.getElementUtils();
     typeUtils = processingEnv.getTypeUtils();
-
     ExternalProvider.registerModuleProvidedTypes(providedTypes);
     providedTypes.addAll(moduleFileProvided);
+    useAvajeConfig =  element("io.avaje.config.Config") != null;
```

Comment:
I think we likely add an extra RequiresPropertyPlugin interface to give us 1 more level of indirection for the properties checks